### ISSUE TITLE
[Bug]: Quantization does not lead to Throughput Speedup (Please Help)

### ISSUE THREAD

**Issue by jvonrad** [vllm-project/vllm#21236]
## Problem

I quantized the llama3-8b myself using gptq, also downloaded the quantized hf version and also used vllm llm compressor and despite it saying that it uses the gptq_marlin kernels for all those I don't see a speedup in neither the awq nor the gptq quantized versions... (running on SM90 H100GPU with Cuda-12.2 and vllm-0.9.2 installed - i pulled newest main version) I also tried different input and output-len, as well as many more prompts like 1000, but its always pretty much the same like the float16 unquantized version.

## my setup:
(vllm) [gwb082@mlcbm002 vllm]$ python -c "import torch; print('Torch:', torch.__version__, '\nCUDA toolkit version:', torch.version.cuda, '\nCUDA available:', torch.cuda.is_available())"
Torch: 2.7.1+cu126 
CUDA toolkit version: 12.6 
CUDA available: True
(vllm) [gwb082@mlcbm002 vllm]$ vllm --version
INFO 07-20 01:56:52 [__init__.py:235] Automatically detected platform cuda.
0.9.2rc2.dev368+g2b504eb77.precompiled

## I am testing like this:

```
(vllm) [gwb082@mlcbm002 vllm]$ vllm bench throughput   --model "/Path/to/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16"   --backend vllm   --dataset-name random   --input-len 32 --output-len 128 --num-prompts 5   --dtype auto   --output-json llama3-8b_fp16.json
````
and despite logging:
"INFO 07-20 01:26:42 [gptq_marlin.py:266] Using MacheteLinearKernel for GPTQMarlinLinearMethod
INFO 07-20 01:26:42 [cuda.py:293] Using Flash Attention backend on V1 engine."

## Results
get this as results:
> Throughput: 5.49 requests/s, 878.94 total tokens/s, 703.16 output tokens/s

and for the unquantized:
> Throughput: 4.76 requests/s, 761.36 total tokens/s, 609.09 output tokens/s

## Entire Logs
This is the entire log for the quantized:

´´´´
(vllm) [gwb082@mlcbm002 vllm]$ vllm bench throughput   --model "/Path/to/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16"   --backend vllm   --dataset-name random   --input-len 32 --output-len 128 --num-prompts 5   --dtype auto   --output-json llama3-8b_fp16.json
INFO 07-20 01:25:59 [__init__.py:235] Automatically detected platform cuda.
When dataset path is not set, it will default to random dataset
INFO 07-20 01:26:09 [datasets.py:355] Sampling input_len from [31, 31] and output_len from [128, 128]
INFO 07-20 01:26:24 [config.py:1566] Using max model len 8192
INFO 07-20 01:26:24 [gptq_marlin.py:170] The model is convertible to gptq_marlin during runtime. Using gptq_marlin kernel.
INFO 07-20 01:26:24 [config.py:2378] Chunked prefill is enabled with max_num_batched_tokens=16384.
INFO 07-20 01:26:32 [__init__.py:235] Automatically detected platform cuda.
INFO 07-20 01:26:38 [core.py:553] Waiting for init message from front-end.
INFO 07-20 01:26:38 [core.py:71] Initializing a V1 LLM engine (v0.9.2rc2.dev368+g2b504eb77) with config: model='/Path/to/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16', speculative_config=None, tokenizer='/home/geiger/gwb082/Jonathans_Thesis/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.float16, max_seq_len=8192, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=gptq_marlin, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detailed_traces=None), seed=0, served_model_name=/home/geiger/gwb082/Jonathans_Thesis/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=True, chunked_prefill_enabled=True, use_async_output_proc=True, pooler_config=None, compilation_config={"level":3,"debug_dump_path":"","cache_dir":"","backend":"","custom_ops":[],"splitting_ops":["vllm.unified_attention","vllm.unified_attention_with_output","vllm.mamba_mixer2"],"use_inductor":true,"compile_sizes":[],"inductor_compile_config":{"enable_auto_functionalized_v2":false},"inductor_passes":{},"use_cudagraph":true,"cudagraph_num_of_warmups":1,"cudagraph_capture_sizes":[512,504,496,488,480,472,464,456,448,440,432,424,416,408,400,392,384,376,368,360,352,344,336,328,320,312,304,296,288,280,272,264,256,248,240,232,224,216,208,200,192,184,176,168,160,152,144,136,128,120,112,104,96,88,80,72,64,56,48,40,32,24,16,8,4,2,1],"cudagraph_copy_inputs":false,"full_cuda_graph":false,"max_capture_size":512,"local_cache_dir":null}
INFO 07-20 01:26:42 [parallel_state.py:1090] rank 0 in world size 1 is assigned as DP rank 0, PP rank 0, TP rank 0, EP rank 0
WARNING 07-20 01:26:42 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
INFO 07-20 01:26:42 [gpu_model_runner.py:1800] Starting to load model /home/geiger/gwb082/Jonathans_Thesis/compressed-models/quantized/Meta-Llama-3-8B-Instruct-quantized.w4a16...
INFO 07-20 01:26:42 [gpu_model_runner.py:1833] Loading model from scratch...
INFO 07-20 01:26:42 [gptq_marlin.py:266] Using MacheteLinearKernel for GPTQMarlinLinearMethod
INFO 07-20 01:26:42 [cuda.py:293] Using Flash Attention backend on V1 engine.
Loading safetensors checkpoint shards:   0% Completed | 0/1 [00:00<?, ?it/s]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:01<00:00,  1.36s/it]
Loading safetensors checkpoint shards: 100% Completed | 1/1 [00:01<00:00,  1.36s/it]

INFO 07-20 01:26:44 [default_loader.py:262] Loading weights took 1.44 seconds
INFO 07-20 01:26:45 [gpu_model_runner.py:1857] Model loading took 5.3507 GiB and 2.279803 seconds
INFO 07-20 01:26:53 [backends.py:530] Using cache directory: /home/geiger/gwb082/.cache/vllm/torch_compile_cache/b01c8904c2/rank_0_0/backbone for vLLM's torch.compile
INFO 07-20 01:26:53 [backends.py:541] Dynamo bytecode transform time: 7.78 s
INFO 07-20 01:26:56 [backends.py:194] Cache the graph for dynamic shape for later use
INFO 07-20 01:28:59 [backends.py:215] Compiling a graph for dynamic shape takes 125.87 s
INFO 07-20 01:29:06 [monitor.py:34] torch.compile takes 133.65 s in total
INFO 07-20 01:29:14 [gpu_worker.py:245] Available KV cache memory: 60.96 GiB
INFO 07-20 01:29:14 [kv_cache_utils.py:775] GPU KV cache size: 499,360 tokens
INFO 07-20 01:29:14 [kv_cache_utils.py:779] Maximum concurrency for 8,192 tokens per request: 60.96x
Capturing CUDA graph shapes: 100%|██████████████████████████████████████████████████████████████████████████████| 67/67 [01:18<00:00,  1.18s/it]
INFO 07-20 01:30:33 [gpu_model_runner.py:2403] Graph capturing finished in 79 secs, took 2.29 GiB
INFO 07-20 01:30:34 [core.py:193] init engine (profile, create kv cache, warmup model) took 228.41 seconds
Adding requests: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 1950.11it/s]
Processed prompts: 100%|██████████████████████████████████| 5/5 [00:00<00:00,  5.53it/s, est. speed input: 177.04 toks/s, output: 708.14 toks/s]
[rank0]:[W720 01:30:36.730351671 ProcessGroupNCCL.cpp:1479] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
Throughput: 5.49 requests/s, 878.94 total tokens/s, 703.16 output tokens/s
Total num prompt tokens:  160
Total num output tokens:  640
´´´´

and this is the entire log for the not-quantized:

´´´´
(vllm) [gwb082@mlcbm002 vllm]$ vllm bench throughput   --model "/home/geiger/gwb082/LLMs/llama-3/Llama-3.1-8B-Instruct"   --backend vllm   --dat
aset-name random   --input-len 32 --output-len 128 --num-prompts 5   --dtype auto   --output-json llama3-8b_fp16.json
INFO 07-20 01:43:19 [__init__.py:235] Automatically detected platform cuda.
When dataset path is not set, it will default to random dataset
INFO 07-20 01:43:29 [datasets.py:355] Sampling input_len from [31, 31] and output_len from [128, 128]
INFO 07-20 01:43:43 [config.py:1566] Using max model len 131072
INFO 07-20 01:43:43 [config.py:2378] Chunked prefill is enabled with max_num_batched_tokens=16384.
INFO 07-20 01:43:51 [__init__.py:235] Automatically detected platform cuda.
INFO 07-20 01:43:54 [core.py:553] Waiting for init message from front-end.
INFO 07-20 01:43:54 [core.py:71] Initializing a V1 LLM engine (v0.9.2rc2.dev368+g2b504eb77) with config: model='/home/geiger/gwb082/LLMs/llama-3/Llama-3.1-8B-Instruct', speculative_config=None, tokenizer='/home/geiger/gwb082/LLMs/llama-3/Llama-3.1-8B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config={}, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=131072, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(backend='auto', disable_fallback=False, disable_any_whitespace=False, disable_additional_properties=False, reasoning_backend=''), observability_config=ObservabilityConfig(show_hidden_metrics_for_version=None, otlp_traces_endpoint=None, collect_detailed_traces=None), seed=0, served_model_name=/home/geiger/gwb082/LLMs/llama-3/Llama-3.1-8B-Instruct, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=True, chunked_prefill_enabled=True, use_async_output_proc=True, pooler_config=None, compilation_config={"level":3,"debug_dump_path":"","cache_dir":"","backend":"","custom_ops":[],"splitting_ops":["vllm.unified_attention","vllm.unified_attention_with_output","vllm.mamba_mixer2"],"use_inductor":true,"compile_sizes":[],"inductor_compile_config":{"enable_auto_functionalized_v2":false},"inductor_passes":{},"use_cudagraph":true,"cudagraph_num_of_warmups":1,"cudagraph_capture_sizes":[512,504,496,488,480,472,464,456,448,440,432,424,416,408,400,392,384,376,368,360,352,344,336,328,320,312,304,296,288,280,272,264,256,248,240,232,224,216,208,200,192,184,176,168,160,152,144,136,128,120,112,104,96,88,80,72,64,56,48,40,32,24,16,8,4,2,1],"cudagraph_copy_inputs":false,"full_cuda_graph":false,"max_capture_size":512,"local_cache_dir":null}
INFO 07-20 01:43:55 [parallel_state.py:1090] rank 0 in world size 1 is assigned as DP rank 0, PP rank 0, TP rank 0, EP rank 0
WARNING 07-20 01:43:55 [topk_topp_sampler.py:59] FlashInfer is not available. Falling back to the PyTorch-native implementation of top-p & top-k sampling. For the best performance, please install FlashInfer.
INFO 07-20 01:43:55 [gpu_model_runner.py:1800] Starting to load model /home/geiger/gwb082/LLMs/llama-3/Llama-3.1-8B-Instruct...
INFO 07-20 01:43:56 [gpu_model_runner.py:1833] Loading model from scratch...
INFO 07-20 01:43:56 [cuda.py:293] Using Flash Attention backend on V1 engine.
Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:00<00:00,  3.95it/s]
Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:01<00:01,  1.28it/s]
Loading safetensors checkpoint shards:  75% Completed | 3/4 [00:02<00:00,  1.07it/s]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:03<00:00,  1.01it/s]
Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:03<00:00,  1.11it/s]

INFO 07-20 01:44:00 [default_loader.py:262] Loading weights took 3.67 seconds
INFO 07-20 01:44:00 [gpu_model_runner.py:1857] Model loading took 14.9889 GiB and 3.915053 seconds
INFO 07-20 01:44:11 [backends.py:530] Using cache directory: /home/geiger/gwb082/.cache/vllm/torch_compile_cache/9efca6e6b5/rank_0_0/backbone for vLLM's torch.compile
INFO 07-20 01:44:11 [backends.py:541] Dynamo bytecode transform time: 6.90 s
INFO 07-20 01:44:17 [backends.py:194] Cache the graph for dynamic shape for later use
INFO 07-20 01:44:54 [backends.py:215] Compiling a graph for dynamic shape takes 42.48 s
INFO 07-20 01:44:59 [monitor.py:34] torch.compile takes 49.37 s in total
INFO 07-20 01:45:01 [gpu_worker.py:245] Available KV cache memory: 51.32 GiB
INFO 07-20 01:45:01 [kv_cache_utils.py:775] GPU KV cache size: 420,416 tokens
INFO 07-20 01:45:01 [kv_cache_utils.py:779] Maximum concurrency for 131,072 tokens per request: 3.21x
Capturing CUDA graph shapes: 100%|██████████████████████████████████████████████████████████████████████████████| 67/67 [01:12<00:00,  1.08s/it]
INFO 07-20 01:46:14 [gpu_model_runner.py:2403] Graph capturing finished in 73 secs, took 1.89 GiB
INFO 07-20 01:46:14 [core.py:193] init engine (profile, create kv cache, warmup model) took 133.29 seconds
Adding requests: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 2586.52it/s]
Processed prompts: 100%|██████████████████████████████████| 5/5 [00:01<00:00,  4.78it/s, est. speed input: 152.88 toks/s, output: 611.51 toks/s]
[rank0]:[W720 01:46:16.023896036 ProcessGroupNCCL.cpp:1479] Warning: WARNING: destroy_process_group() was not called before program exit, which can leak resources. For more info, please see https://pytorch.org/docs/stable/distributed.html#shutdown (function operator())
Throughput: 4.76 requests/s, 761.36 total tokens/s, 609.09 output tokens/s
Total num prompt tokens:  160
Total num output tokens:  640
´´´´

### Your current environment (if you think it is necessary)

```text
The output of `python collect_env.py`
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by QuditWolf** [comment]
WARNING 07-20 05:02:23 [config.py:1046] gguf quantization is not fully optimized yet. The speed can be slower than non-quantized models.           

https://docs.vllm.ai/en/v0.9.0/features/quantization/gguf.html

The project page does state that gguf is under-optimzed at the moment

**Comment by jonny-vr** [comment]
> WARNING 07-20 05:02:23 [config.py:1046] gguf quantization is not fully optimized yet. The speed can be slower than non-quantized models.
> 
> https://docs.vllm.ai/en/v0.9.0/features/quantization/gguf.html
> 
> The project page does state that gguf is under-optimzed at the moment

this is not gguf, but gptq quantization

**Comment by ZhuYe0729** [comment]
Hello, I also encountered a similar problem. Could you share with me how you solved it?
